lib/cloudregister/registerutils.py

```diff
@@ -2042,3 +2279,118 @@ def __mv_file_backup(filename):
     message = 'File not preserved' if failed else 'File preserved'
     logging.info(message)
     return failed
+
+
+# ----------------------------------------------------------------------------
+def __set_registries_conf_podman(private_registry_fqdn):
+    """Set the registry search order for Podman."""
+    registries_conf = {}
+    if os.path.exists(REGISTRIES_CONF_PATH):
+        registries_conf, failed = get_registry_conf_file(
+            REGISTRIES_CONF_PATH, 'podman'
+        )
+        if failed:
+            return False
+
+    public_registry_fqdn = 'registry.suse.com'
+    unqualified_search_reg = []
+    unqualified_search_reg = registries_conf.get(
+        'unqualified-search-registries', []
+    )
+    modified = False
+    if unqualified_search_reg:
+        priv_index = -1
+        pub_index = -1
+        if private_registry_fqdn in unqualified_search_reg:
+            priv_index = unqualified_search_reg.index(private_registry_fqdn)
+        if public_registry_fqdn in unqualified_search_reg:
+            pub_index = unqualified_search_reg.index(public_registry_fqdn)
+
+        if not priv_index == 0 or not pub_index == 1:
+            if priv_index > 0:
+                unqualified_search_reg.pop(priv_index)
+            if pub_index > 0:
+                unqualified_search_reg.pop(pub_index)
+            modified = True
```

Comment:
This logic in the ```__set_``` method drops an entry under certain conditions prior adding data. I wonder if it would be possible to just call ```clean_registries_conf_podman``` in ```__set_registries_conf_podman``` and keep only one code path that actually drops entries ?

Reply:
so instead of 

- open file
- check content
- modify (drop and add) content if needed
- write file

you suggest

- open file 
- drop content
- write file
- open file
- add content
- write file
?

Reply:
having one code path to drop entries seems fine to me, just wonder if it is not that cheap ?

Reply:
I was making the suggestion just for logic and consistency reasons. I agree with you that your current implementation is more performant. The part that matters most should be followed and I leave this decision up to you. I'm fine either way. In this particular case I believe less code makes it easier to prevent pitfalls due to the many ```if``` conditions and I think performance is not too much important at this point. So I looked at it more from a transaction perspective in a way that you can set what is needed when you start from a clean state

Reply:
At first glance my gut feel is that the logic can be simplified. But I do not have enough peace an quiet to think about what that would look like. We will need some refactoring paths over the code anyway. As such I'd tend to leave this as is at the moment.